Initial LVGL doc for ESPHome (#3678)
OttoWinter/esphomedocs: components/switch/lvgl.rst

.. _lvgl-swi:

LVGL Switch
===========

.. seo::

>     :description: Instructions for setting up a LVGL widget switch.

Reviewer: ```suggestion
    :description: Instructions for setting up an LVGL widget switch.
```

    :image: ../images/lvgl_c_swi.png

> The ``lvgl`` switch platform creates a switch from a LVGL widget

Reviewer: ```suggestion
The ``lvgl`` switch platform creates a switch from an LVGL widget
```

and requires :ref:`LVGL <lvgl-main>` to be configured.

> Supported widgets are :ref:`lvgl-wgt-btn` (with ``checkable`` option enabled), :ref:`lvgl-wgt-swi` and :ref:`lvgl-wgt-chk`. A single switch supports a single widget, thus you need to choose among which one's state you want to use.

Reviewer: ```suggestion
Supported widgets are :ref:`lvgl-wgt-btn` (with ``checkable`` option enabled), :ref:`lvgl-wgt-swi` and :ref:`lvgl-wgt-chk`.
There is a one-to-one relationship between any given widget and its associated switch component.
```

Configuration options:
----------------------

- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.

> - **name** (**Required**, string): The name of the switch.

Reviewer: ```suggestion
```

Example:

.. code-block:: yaml

    switch:
      - platform: lvgl
        widget: checkbox_id
        name: LVGL switch

> Check out :ref:`lvgl-cook-outbin` in the Cookbook for an example how to set up a LVGL Switch component to interact directly with a GPIO.

Reviewer: ```suggestion
Check out :ref:`lvgl-cook-outbin` in the Cookbook for an example illustrating how to set up a LVGL Switch component to interact directly with a GPIO.
```
